IPython/core/magics/osm.py

```diff
@@ -31,6 +31,51 @@ class OSMagics(Magics):
     """Magics to interact with the underlying OS (shell-type functionality).
     """
 
+    def __init__(self, shell=None, **kwargs):
+
+        # Now define isexec in a cross platform manner.
+        self.is_posix = False
+        self.execre = None
+        if os.name == 'posix':
+            self.is_posix = True
+        else:
+            try:
+                winext = os.environ['pathext'].replace(';','|').replace('.','')
+            except KeyError:
+                winext = 'exe|com|bat|py'
+            
+            self.execre = re.compile(r'(.*)\.(%s)$' % winext,re.IGNORECASE)
+
+        # call up the chain
+        super(OSMagics, self).__init__(shell=shell, **kwargs)
```

Comment:
This can be just `super()` since IPython requires Python 3

Reply:
thanks. I will update this